DigitalLearningSolutions.Data/DataServices/RegistrationDataService.cs

```diff
@@ -281,16 +290,21 @@ OUTPUT Inserted.ID
             int centreId,
             string? centreSpecificEmail,
             int userId,
+            bool isSelfRegistered,
             IDbTransaction transaction
         )
         {
             if (!string.IsNullOrWhiteSpace(centreSpecificEmail))
             {
+                // TODO: 915 - replace false with call to IsEmailVerifiedForUser
+                var emailIsVerifiedForUser = false;
+                var emailShouldBeVerified = emailIsVerifiedForUser || !isSelfRegistered;

```

Comment:
Because this is a data service, I'd like it to be dumb as possible and pull logic into the Registration Service. Can just pass `bool emailIsVerified` in as a param?

Reply:
Moved the logic into the service methods that call this, it caused quite a lot of ripple effects since it's used in a bunch of methods